rascal-vscode-extension/src/test/vscode-suite/dsl.test.ts

```diff
@@ -53,6 +53,18 @@ parameterizedDescribe(function (errorRecovery: boolean) {
         const repl = new RascalREPL(bench, driver);
         await repl.start();
         await repl.execute("import demo::lang::pico::OldStyleLanguageServer;");
+
+        // If Pico was registered before as part of another series of tests,
+        // then it needs to be unregistered first (because error recovery
+        // en/disabledness affects which contributors to use). Until issue #630
+        // is fixed (race between `unregister` and `register`), the
+        // unregistration can't reliably be done as part of `main` (tried in
+        // commit `a955a05`). Instead, it's done here and followed by a suitably
+        // long sleep.
+        await repl.execute("import util::LanguageServer;");
```

Comment:
This seems like a decent temporary fix. Have you investigated what a permanent solution could look like?